### PR TITLE
Update CSM tutorial

### PR DESCRIPTION
**Pull request by vhirtham**
## Changes

Updates the tutorial of the CSM.

## Checks

- [x] updated CHANGELOG.rst
- [x] ~~updated tests~~
- [x] updated doc/
- [x] update example/tutorial notebooks
- [x] ~~update manifest file~~

### CODE REVIEW

**Comment by marscher** [comment]
Thank you for updating the tutorial. It was a joy reading :+1: 

I will commit some changes (e.g. the parameter enumeration). 
Another thing I've noticed is that the example about subsystems uses unrealistic units for the torch position in relation to the robot head (2 mm). This is perhaps a minor detail, but I guess we should change it to "cm" or increase the numbers to match a more realistic point of view. What do you think?

**Comment by marscher** [comment]
I've also added some padding for the csm.plot_graph method, so the labels of the subsystems are readable (they used to be truncated on the right side).

File: tutorials/transformations_02_coordinate_system_manager.ipynb [hunk 1]
```diff
@@ -9,18 +9,6 @@
     "## Imports"
```
Comment (marscher) [inline tutorials/transformations_02_coordinate_system_manager.ipynb:20]: <p>If this tutorial is meant to replace the "transformations_0*_tutorials", I'd not place this note into the introduction. At least do not make it the first sentence. Start with "The purpose of the CSM is...".</p><p> </p><p>Would it be possible to link the CSM to its docstring inside the notebook?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BAMWelDX/weldx/pull/767/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (marscher) [inline tutorials/transformations_02_coordinate_system_manager.ipynb:20]: <p>a "member function" is a method, right?</p><p>I'd use a enumeration for the parameters, to make the section more readable.</p><p> </p><p>Please explain the concept of a reference system before using it for the parameter explanation. E.g. show an equation to make things clear or just write one or two sentences what's the actual relation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BAMWelDX/weldx/pull/767/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (marscher) [inline tutorials/transformations_02_coordinate_system_manager.ipynb:20]: <p>The overview you've just given, right? The link contains ALL the details (which might be overwhelming, if this is the starting point).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BAMWelDX/weldx/pull/767/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (marscher) [inline tutorials/transformations_02_coordinate_system_manager.ipynb:20]: <p>not only internally, but the logical relation of the systems is a tree. Eventually you can elaborate on that (1-2 sentences).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BAMWelDX/weldx/pull/767/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (marscher) [inline tutorials/transformations_02_coordinate_system_manager.ipynb:20]: <p>Nowadays, we call it Jupyter notebook :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BAMWelDX/weldx/pull/767/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

**Comment by vhirtham** [comment]
> Thank you for updating the tutorial. It was a joy reading 👍
> 
> I will commit some changes (e.g. the parameter enumeration). Another thing I've noticed is that the example about subsystems uses unrealistic units for the torch position in relation to the robot head (2 mm). This is perhaps a minor detail, but I guess we should change it to "cm" or increase the numbers to match a more realistic point of view. What do you think?

Yes, that sounds reasonable. The main focus here was to get everything "into the camera". However, this was more problematic when the static matplotlib plots were used. I think with k3d, we can experiment a bit more with the values and stretch everything out. Also, the xarray data points in the section about spatial data are barely visible in the plots. I think it might be a good idea to rename the data to "point cloud" instead of "sensor positions" (or whatever it was named) and generate a denser set of data with a list comprehension.

File: tutorials/transformations_02_coordinate_system_manager.ipynb (hunk 1 above)
Comment (vhirtham) [reply tutorials/transformations_02_coordinate_system_manager.ipynb:20]: <p>Not sure what you mean. The LCS tutorial will also get an update and this on isn't meant to replace anything. Its just an update :)</p><p> </p><p>I can change the order though.</p><p> </p><p>We can link the API documentation page.</p>

Comment (vhirtham) [reply tutorials/transformations_02_coordinate_system_manager.ipynb:22]: <p>Changed it.</p>

Comment (vhirtham) [reply tutorials/transformations_02_coordinate_system_manager.ipynb:22]: <p>done</p>

[the remark this answers is not in the document]
Comment (vhirtham) [reply tutorials/transformations_02_coordinate_system_manager.ipynb:22]: <p>yes, automatic links would be nice. I added one manually, but it is rather suboptimal. Unfortunately, code as text in links doesn't work well (bad visibility).</p><p> </p><p>Removed mentioning of factory method.</p>

Comment (vhirtham) [reply tutorials/transformations_02_coordinate_system_manager.ipynb:22]: <p>It's more about the construction methods that aren't explicitly mentioned here. I think it is a managable task for the user to look through the API doc ;) </p>

Comment (vhirtham) [reply tutorials/transformations_02_coordinate_system_manager.ipynb:22]: <p>Think the rendered images should be enough, don't you think?</p>

Comment (vhirtham) [reply tutorials/transformations_02_coordinate_system_manager.ipynb:22]: <p>;) changed it</p>

[the remark this answers is not in the document]
Comment (vhirtham) [reply tutorials/transformations_02_coordinate_system_manager.ipynb:22]: <p>My bad. I changed the note to be more specific. It was about CSM instances that get merged into each other</p>

**Comment by vhirtham** [comment]
Tweaked some values as suggested. Take a look at the new plots and let me know what you think

**Review by marscher — APPROVED** [review]

**Comment by CagtayFabry** [comment]
great, thank you @vhirtham :)

**Pull request merged**